File: frontend/medmatch-ui/src/features/matching/types/matching.ts
export type EligibilityStatus =
  | "Eligible"
  | "Not Eligible"
  | "Possibly Eligible";

export interface MatchingResult {
  id: string;

  trial_id: string;

  title: string;

  condition: string | null;

  phase: string | null;

  status: string | null;

  brief_summary: string | null;

  criteria_type: string;

  description: string;

  distance: number;
}

export interface MatchingResponse {
  query: string;
  total_matches: number;
  returned_matches: number;
  similarity_threshold: number;
  matches: MatchingResult[];
}

export interface EligibilityResponse {
  eligibility:
    | "Eligible"
    | "Not Eligible"
    | "Possibly Eligible";

  confidence: number;

  summary: string;

  matched_inclusion: string[];

  failed_inclusion: string[];

  satisfied_exclusion: string[];

  triggered_exclusion: string[];

  missing_information: string[];

  recommendation: string;

  reasoning: string;
}

export interface MatchingRequest {
  patient_note: string;
  limit: number;
}

File: services/ai-service/app/schemas/matching.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class MatchingResult(BaseModel):
    """
    Single semantic search result.
    """

    id: UUID

    trial_id: UUID

    title: str

    condition: str | None = None

    phase: str | None = None

    status: str | None = None

    brief_summary: str | None = None

    criteria_type: str

    description: str

    distance: float = Field(
        ge=0.0,
        le=2.0,
        description="Cosine distance between embeddings.",
    )

    model_config = ConfigDict(
        from_attributes=True,
    )


class MatchingResponse(BaseModel):
    """
    Semantic search response.
    """

    query: str = Field(
        description="Patient note used for semantic retrieval.",
    )

    total_matches: int = Field(
        ge=0,
        description="Total number of unique matching trials found.",
    )

    returned_matches: int = Field(
        ge=0,
        description="Number of unique trial matches returned.",
    )

    similarity_threshold: float = Field(
        ge=0.0,
        le=2.0,
        description="Maximum cosine distance accepted.",
    )

    matches: list[MatchingResult]

    model_config = ConfigDict(
        from_attributes=True,
    )